v01/http.go

```diff
@@ -97,8 +97,8 @@ func (e *Event) UnmarshalBinary(req *http.Request) error {
 			continue
 		}
 
-		k = strings.TrimLeft(k, "Ce-")
-		k = strings.TrimLeft(k, "X-")
+		k = strings.TrimPrefix(k, "Ce-")
```

Comment:
Given that TrimPrefix does an "if" to see if "Ce-" is there, and we're doing that on line 96, is there any reason we're not just using `k[3:]` and using `Ce-` on line 96?  Actually 96 is wrong, it should have a dash.

Reply:
In the larger scope, this loop is trying to operate on only header values that are marked as CloudEvent headers. If the header is not a CloudEvent header, it is ignored. If it is a CloudEvent header then I extract the raw property name and compare it to the well known CloudEvent fields to set that field on the Event object. If the raw property name is not in the map of known Event fields then it gets stuck in the (internal) extensions bag.

You are right, line 96 is missing a dash.

Reply:
yeah I get all of that, I just think we're doing it the hard way :-)
see: https://github.com/cloudevents/sdk-go/pull/4#discussion_r231731994  too